Mathlib/Algebra/MonoidAlgebra/Division.lean

```diff
@@ -190,6 +197,28 @@ theorem modOf_add_divOf (x : AddMonoidAlgebra k G) (g : G) : x %ᵒᶠ g + of' k
   by rw [add_comm, divOf_add_modOf]
 #align add_monoid_algebra.mod_of_add_div_of AddMonoidAlgebra.modOf_add_divOf
 
+lemma neg_modOf (x : AddMonoidAlgebra k' G) (g : G) : (-x) %ᵒᶠ g = - (x %ᵒᶠ g) := by
+  have eq1 : - _ = _ := (congr_arg (- .) $ modOf_add_divOf x g).trans
+    (modOf_add_divOf (-x) g).symm
+  simp only [neg_add_rev, neg_divOf, mul_neg, add_comm] at eq1
+  rwa [add_left_inj, eq_comm] at eq1
+
+-- Need `k''` to find commutativity
+lemma add_modOf (x y : AddMonoidAlgebra k'' G) (g : G) : (x + y) %ᵒᶠ g = x %ᵒᶠ g + y %ᵒᶠ g := by
+  have eq1 : _ + _ = _ + _ := (congr_arg₂ (. + .) (modOf_add_divOf x g) (modOf_add_divOf y g)).trans
+    (modOf_add_divOf (x + y) g).symm
+  rwa [add_divOf, mul_add, show ∀ (a b c d), a + b + (c + d) = a + c + b + d from ?_, add_assoc,
+    add_left_inj, eq_comm] at eq1
+  intros; ring
```

Comment:
```suggestion
lemma add_modOf [IsCancelAdd k] (x y : AddMonoidAlgebra k G) (g : G) : (x + y) %ᵒᶠ g = x %ᵒᶠ g + y %ᵒᶠ g := by
  have eq1 : _ + _ = _ + _ := (congr_arg₂ (. + .) (modOf_add_divOf x g) (modOf_add_divOf y g)).trans
    (modOf_add_divOf (x + y) g).symm
  letI missing : IsCancelAdd (G →₀ k) := {
    add_left_cancel := fun a b c h => Finsupp.ext <| fun x => add_left_cancel (FunLike.congr_fun h x)
    add_right_cancel := fun a b c h => Finsupp.ext <| fun x => add_right_cancel (FunLike.congr_fun h x)
  }
  letI also_missing : IsCancelAdd (AddMonoidAlgebra k G) := missing
  rwa [add_divOf, mul_add, add_add_add_comm, add_left_inj, eq_comm] at eq1
```
where those two `letI`s should be global instances

Reply:
I have moved them outside